docker-setup.sh

```diff
@@ -0,0 +1,665 @@
+#!/bin/bash
+
+set -e
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+IMAGE_NAME="coinswap/coinswap"
+BITCOIN_IMAGE_NAME="coinswap/bitcoin-mutinynet"
+CONFIG_FILE="$SCRIPT_DIR/.docker-config"
+VERSION_FILE="$SCRIPT_DIR/docker/bitcoin-mutinynet.version"
+
+DEFAULT_BITCOIN_DATADIR="/home/coinswap/.bitcoin"
+DEFAULT_BITCOIN_NETWORK="signet"
+DEFAULT_BITCOIN_RPC_PORT="38332"
+DEFAULT_BITCOIN_ZMQ_PORT="28332"
+DEFAULT_MAKERD_PORT="6102"
+DEFAULT_MAKERD_RPC_PORT="6103"
+DEFAULT_TOR_SOCKS_PORT="9050"
+DEFAULT_TOR_CONTROL_PORT="9051"
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m'
+print_info() {
+    echo -e "${BLUE}[INFO]${NC} $1"
+}
+
+print_success() {
+    echo -e "${GREEN}[SUCCESS]${NC} $1"
+}
+
+print_warning() {
+    echo -e "${YELLOW}[WARNING]${NC} $1"
+}
+
+print_error() {
+    echo -e "${RED}[ERROR]${NC} $1"
+}
+
+load_config() {
+    if [ -f "$CONFIG_FILE" ]; then
+        source "$CONFIG_FILE"
+        print_info "Loaded configuration from $CONFIG_FILE"
+    fi
+}
+
+save_config() {
+    cat > "$CONFIG_FILE" << EOF
+# Coinswap Docker Configuration
+BITCOIN_DATADIR="$BITCOIN_DATADIR"
+BITCOIN_NETWORK="$BITCOIN_NETWORK"
+BITCOIN_RPC_PORT="$BITCOIN_RPC_PORT"
+BITCOIN_ZMQ_PORT="$BITCOIN_ZMQ_PORT"
+MAKERD_PORT="$MAKERD_PORT"
+MAKERD_RPC_PORT="$MAKERD_RPC_PORT"
+TOR_SOCKS_PORT="$TOR_SOCKS_PORT"
+TOR_CONTROL_PORT="$TOR_CONTROL_PORT"
+USE_EXTERNAL_BITCOIND="$USE_EXTERNAL_BITCOIND"
+USE_EXTERNAL_TOR="$USE_EXTERNAL_TOR"
+EXTERNAL_BITCOIND_HOST="$EXTERNAL_BITCOIND_HOST"
+EXTERNAL_TOR_HOST="$EXTERNAL_TOR_HOST"
+EOF
+    print_success "Configuration saved to $CONFIG_FILE"
+}
+
+check_port() {
+    local port=$1
+    if nc -z localhost "$port" 2>/dev/null; then
+        return 1
+    else
+        return 0
+    fi
+}
+
+check_tor_running() {
+    if check_port 9050; then
+        return 1
+    else
+        return 0
+    fi
+}
+
+check_bitcoind_running() {
+    local port=${1:-18332}
+    if check_port "$port"; then
+        return 1
+    else
+        return 0
+    fi
+}
+
+configure_setup() {
+    echo ""
+    print_info "Coinswap Docker Configuration"
+    echo "============================================"
+    echo ""
+    
+    print_info "Bitcoin Core Configuration"
+    echo "----------------------------------------"
+    
+    read -p "Bitcoin data directory [${DEFAULT_BITCOIN_DATADIR}]: " input
+    BITCOIN_DATADIR="${input:-$DEFAULT_BITCOIN_DATADIR}"
+    
+    echo ""
+    echo "Select Bitcoin network:"
+    echo "1) signet (default - test network)"
+    echo "2) regtest (for testing)"
+    echo "3) testnet (test network)"  
+    echo "4) mainnet (CAUTION: real bitcoin)"
+    read -p "Network [1]: " network_choice
```

Comment:
We don't have configs for testnet and mainnet. just signet and regtest. And we won't add them anytime soon.

So lets just give 2 options. `Mutinynet`, `Regtest`. Mutinynet will trigger the signet. Theres no point of adding general public signet option either.  It doesn't have enough infra.